src/cypress-helper.ts

```diff
@@ -813,16 +825,18 @@ export class CypressHelper {
      * @param [index]
      */
     elementByTestId: (dataTestID: string, index?: number) =>
-      this.options.handleSlotShadowDOM &&
-      dataTestID.endsWith(`-${this.options.shadowSlotSuffix}`)
-        ? this.get.nthBySelector(dataTestID, index).then(slot =>
-            cy.wrap(
-              Cypress.$(slot as JQuery<HTMLSlotElement>)
-                .get(0)
-                .assignedNodes()[0].parentElement!
-            )
-          )
-        : this.get.nthBySelector(dataTestID, index),
+      this.get
+        .nthBySelector(dataTestID, index)
+        .should(_ => { })
+        .then(elements => {
+          if (elements.length !== 0) {
+            return this.shouldHandleShadowDomSlot(dataTestID)
+              ? this.getShadowSlotElement(dataTestID, index)
+              : this.get.nthBySelector(dataTestID, index)
+          } else {
+            return this.get.nthBySelector(dataTestID, index)
+          }

```

Comment:
maybe we don't need the else?

Reply:
@neneplusdev , maybe we don't need the else, if the inside the "if" there is already a retrun command